File: business/src/main/java/no/nav/sbl/dialogarena/soknadinnsending/business/service/soknadservice/Systemdata.java
package no.nav.sbl.dialogarena.soknadinnsending.business.service.soknadservice;

import no.nav.sosialhjelp.soknad.domain.SoknadUnderArbeid;

public interface Systemdata {

    void updateSystemdataIn(SoknadUnderArbeid soknadUnderArbeid, String token);
    
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/common/systemdata/Systemdata.kt
package no.nav.sosialhjelp.soknad.common.systemdata

import no.nav.sosialhjelp.soknad.domain.SoknadUnderArbeid
import org.springframework.stereotype.Component

interface Systemdata {
    fun updateSystemdataIn(soknadUnderArbeid: SoknadUnderArbeid)
}

@Component
open class SystemdataUpdater(private val systemdatas: List<Systemdata>) {
    open fun update(soknadUnderArbeid: SoknadUnderArbeid) {
        systemdatas.forEach {
            it.updateSystemdataIn(soknadUnderArbeid)
        }
    }
}
